module/VuFind/src/VuFind/Marc/Serialization/MarcInJson.php

```diff
@@ -0,0 +1,202 @@
+ */
+namespace VuFind\Marc\Serialization;
+
+/**
+ * MARC-in-JSON format support class.
+ *
+ * @category VuFind
+ * @package  MARC
+ * @author   Ere Maijala <ere.maijala@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:record_drivers Wiki
+ */
+class MarcInJson extends AbstractSerializationFile implements SerializationInterface
+{
+    /**
+     * Current collection file
+     *
+     * @var array
+     */
+    protected $collection = null;
+
+    /**
+     * Current position in collection file
+     *
+     * @var int
+     */
+    protected $position = 0;
+
+    /**
+     * Check if this class can parse the given MARC string
+     *
+     * @param string $marc MARC
+     *
+     * @return bool
+     */
+    public static function canParse(string $marc): bool
+    {
+        // A pretty naïve check, but it's enough to tell the different formats apart
+        return substr(trim($marc), 0, 1) === '{';
+    }
+
+    /**
+     * Check if the serialization class can parse the given MARC collection string
+     *
+     * @param string $marc MARC
+     *
+     * @return bool
+     */
+    public static function canParseCollection(string $marc): bool
+    {
+        // A pretty naïve check, but it's enough to tell the different formats apart
+        return substr(trim($marc), 0, 1) === '[';
+    }
+
+    /**
+     * Check if the serialization class can parse the given MARC collection file
+     *
+     * @param string $file File name
+     *
+     * @return bool
+     */
+    public static function canParseCollectionFile(string $file): bool
+    {
+        if (false === ($f = fopen($file, 'rb'))) {
+            throw new \Exception("Cannot open file '$file' for reading");
+        }
+        $s = '';
+        do {
+            $s .= fgets($f, 10);
+        } while (strlen(ltrim($s)) < 5 && !feof($f));
+        fclose($f);
+
+        return self::canParseCollection($s);
+    }
+
+    /**
+     * Parse MARC collection from a string into an array
+     *
+     * @param string $collection MARC record collection in the format supported by
+     * the serialization class
+     *
+     * @throws \Exception
+     * @return array
+     */
+    public static function collectionFromString(string $collection): array
+    {
+        return json_decode($collection, true);
+    }
+
+    /**
+     * Parse MARC-in-JSON
+     *
+     * @param string $marc JSON
+     *
+     * @throws \Exception
+     * @return array
+     */
+    public static function fromString(string $marc): array
+    {
+        return json_decode($marc, true);
+    }
+
+    /**
+     * Convert record to ISO2709 string
+     *
+     * @param array $record Record data
+     *
+     * @return string
+     */
+    public static function toString(array $record): string
+    {
+        // We need to cast any subfield with '0' as key to an object to, otherwise
```

Comment:
I think there's an extra "to" in this sentence.

Reply:
Fixed!